internal/sourcemap/body_caching.go

```diff
@@ -55,6 +55,13 @@ func NewBodyCachingFetcher(
 	go func() {
 		logger.Debug("listening for invalidation...")
 
+		// prevent goroutine leak
+		// this can only happens in tests, otherwise
+		// invalidationChan is never null
+		if invalidationChan == nil {
+			return
+		}
+
```

Comment:
Can't we make the channel non nil in the tests? We could pass in a closed channel which would cause the coroutine to shut down immediately.

Reply:
Oh, that works too. Changed :+1: